prometheus/collectors/go_collector_go120_test.go

```diff
@@ -116,3 +118,18 @@ func withSchedulerMetrics() []string {
 func withDebugMetrics() []string {
 	return withBaseMetrics([]string{})
 }
+
+var defaultRuntimeMetrics = []string{
+	"go_sched_gomaxprocs_threads",
+}
+
+func withDefaultRuntimeMetrics(metricNames []string, withoutGC, withoutSched bool) []string {
+	// If withoutSched is true, exclude "go_sched_gomaxprocs_threads".
```

Comment:
Nice, so those tests verify if our default metrics can be removed by "without" option, right? Great!

Reply:
Basically I had to find a way to compose the expected default Go runtime metrics in the tests according to the various allow/deny lists, in order to mirror the logic's behaviour, which allows removing the default runtime metrics for example and which also avoids duplicated metrics